Lib/random.py

```diff
@@ -733,8 +732,8 @@ def random(self):
 
     def getrandbits(self, k):
         """getrandbits(k) -> x.  Generates an int with k random bits."""
-        if k <= 0:
-            raise ValueError('number of bits must be greater than zero')
+        if k < 0:
+            raise ValueError('number of bits must be positive')
```

Comment:
I think you can use EAFP here. `os.urandom()` fails with negative argument.

Reply:
Disagreed. it's much better, readability-wise, to check invalid input explicitly.

Reply:
In this case you can remove the check completely, because the error raised by `os.urandom()` is good enough: "negative argument not allowed". I think it is more Pythonic.

Otherwise you need to replace "positive" with "non-negative" in the error message or rewrite it in other way.

Reply:
I'll use "non-negative" then.